src/stores/SpaceStore.tsx

```diff
@@ -629,11 +629,18 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
     };
 
     private onRoom = (room: Room, newMembership?: string, oldMembership?: string) => {
-        const membership = newMembership || room.getMyMembership();
+        const roomMembership = room.getMyMembership();
+        if (roomMembership === null) {
```

Comment:
might be safer in the long run:
```suggestion
        if (!roomMembership) {
```